File: ginfo.py
#キャラクターカードクラス
class Character:
    def __init__(self,
                 No,
                 image,
                 name,
                 type,
                 color,
                 need_energy,
                 action_point,
                 buttle_point,
                 out_energy,
                 active,
                 damage,
                 impact,
                 unimpact,
                 step,
                 append_effect,
                 main_effect,
                 attack_effect,
                 block_effect,
                 trash_effect,
                 never_effect,
                 trigger_effect):
        
        #基礎ステータス
        self.No = No
        self.image = image
        self.name = name
        self.type = type
        self.color = color
        self.need_energy = need_energy
        self.actioin_point = action_point
        self.buttle_point = buttle_point
        self.out_energy = out_energy
        self.active = active
        self.damage = damage

        #キーワード効果
        self.impact = impact
        self.unimpact = unimpact
        self.step = step

        #カード効果
        self.append_effect = append_effect
        self.main_effect = main_effect
        self.attack_effect = attack_effect
        self.block_effect = block_effect
        self.trash_effect = trash_effect
        self.never_efect = never_effect
        self.trriger_efffect = trigger_effect

#呪術廻戦/青/キャラクター
JJK036 = Character("JJK036",None,"伊地知 潔高",None,"Blue",0,1,1000,1,False,1,0,False,False,None,None,None,None,None,None,None)
JJK037 = Character("JJK037",None,"虎杖 悠仁",None,"Blue",0,1,1500,1,False,1,0,False,False,None,None,None,None,None,None,None)
JJK038 = Character("JJK038",None,"虎杖 悠仁",None,"Blue",2,1,3000,1,False,1,0,False,False,None,None,None,None,None,None,None)
JJK043 = Character("JJK043",None,"釘崎 野薔薇",None,"Blue",0,1,1500,1,False,1,0,False,False,None,None,None,None,None,None,None)

#呪術廻戦/黄/キャラクター
JJK001 = Character("JJK001",None,"虎杖 悠仁",None,"Yellow",0,1,1000,1,False,1,0,False,False,None,None,None,None,None,None,None)
JJK006 = Character("JJK006",None,"釘崎 野薔薇",None,"Yellow",0,1,1500,1,False,1,0,False,False,None,None,None,None,None,None,None)
JJK024 = Character("JJK024",None,"玉犬:黒＆白","式神","Yellow",0,1,2000,1,False,1,0,False,False,None,None,None,None,None,None,None)


#呪術廻戦リスト
Blue_JJK = ["呪術廻戦/青",JJK036,JJK037,JJK038,JJK043]
Yellow_JJK = ["呪術廻戦/黄",JJK001,JJK006,JJK024]

File: ginfo.rb
#カード全般クラス
class Cards
	#基本プロパティ
	attr_accessor :No
	attr_accessor :type
	attr_accessor :image
	attr_accessor :name
	attr_accessor :color
	attr_accessor :summon_E
	attr_accessor :summon_AP

	#カード効果
	attr_accessor :trigger_EF
	attr_accessor :hand_never_EF

	def initialize(No:,type:,image:,name:,color:,summon_E:,summon_AP:,trigger_EF:nil,hand_never_EF:nil)
		 @No = No
		 @type = type
		 @image = image
		 @name = name
		 @color = color
		 @summon_E = summon_E
		 @summon_AP = summon_AP
		 @trigger_EF = trigger_EF
		 @hand_never_EF = hand_never_EF
	end
end

#キャラクターカードクラス
class Character < Cards
	#基礎ステータス
	attr_accessor :attribute
	attr_accessor :out_E
	attr_accessor :BP
	attr_accessor :damage

	#キーワード効果
	attr_accessor :impact
	attr_accessor :unimpact
	attr_accessor :step

	#カード効果
	attr_accessor :append_EF
	attr_accessor :main_EF
	attr_accessor :attack_EF
	attr_accessor :block_EF
	attr_accessor :void_EF
	attr_accessor :line_never_EF

	def initialize(No:,type:,image:,name:,color:,summon_E:,summon_AP:,trriger_EF:nil,hand_never_EF:nil,attribute:nil,out_E:,BP:,damage:,impact:0,unimpact:false,step:false,append_EF:nil,main_EF:nil,attack_EF:nil,block_EF:nil,void_EF:nil,line_never_EF:nil)
		super(No:No,type:type,image:image,name:name,color:color,summon_E:summon_E,summon_AP:summon_AP,trriger_EF:trigger_EF,hand_never_EF:hand_never_EF)
		@attribute = attribute
		@out_E = out_E
		@BP = BP
		@damage = damage
		@impact = impact
		@unimpact = unimpact
		@step = step
		@append_EF = append_EF
		@main_EF = main_EF
		@attack_EF = attack_EF
		@block_EF = block_EF
		@void_EF = void_EF
		@line_never_EF = line_never_EF
	end

end
